src/components/App.tsx

```diff
@@ -104,7 +136,103 @@ function App() {
       });
   };
 
-  // Logic to save cards will go here on backend setup
+  // Checking for token
+  useEffect(() => {
+    // Setting jwt to the token from local storage
+    const jwt = localStorage.getItem("jwt");
+    if (jwt) {
+      localStorage.setItem("jwt", jwt);
+      auth
+        .checkToken(jwt)
+        .then((res) => {
+          setIsLoggedIn(true);
+          setCurrentUser(res.data);
+          getSavedArticles(jwt).then((SavedArticles) => {
+            setSavedNewsArticles(SavedArticles);
+          });
+        })
+        .catch((err) => {
+          console.error(err);
+        });
+    }
+  }, []);
+
+  // Logic to login an existing user
+  function handleLogin({ email, password }: loginProps) {
+    auth
+      .authorize({ email, password })
+      .then((res) => {
+        if (res) {
+          localStorage.setItem("jwt", res.token);
+          auth
+            .checkToken(res.token)
+            .then((data) => {
+              setCurrentUser(data.data);
+              setIsLoggedIn(true);
+              getSavedArticles(res.token).then((SavedArticles) => {
+                setSavedNewsArticles(SavedArticles);
+              });
+            })
+            .catch((err) => {
+              console.error(err);
+            });
+        }
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+
+  // Logic to register a new user
+  function handleSignup({ email, password, avatar, name }: signupProps) {
+    auth
+      .registration({ name, avatar, email, password })
+      .then((res) => {
+        if (res) {
+          handleLogin({ email, password });
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  // Logic go update the profile
+  const updateProfile = ({ name, avatar }: updateUserProps) => {
+    updateUser({ name, avatar })
+      .then(({ data }) => {
+        setCurrentUser(data);
+        return data;
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+
+  const handleSaveArticle = (card: Article) => {
+    saveArticle(card)
+      .then((response) => response.json())
+      .then(({ data }) => {
+        setSavedNewsArticles([...savedNewsArticles, data]);
+        setSelectedArticleId(data._id);
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors